lib/ruleHelper.js

```diff
@@ -163,7 +163,9 @@ RuleHelper.prototype = {
 
             // the `init` property carries the right-hand side of the variable definition:
             const varInitAs = def.node.init;
-            if (!this.allowedExpression(varInitAs, escapeObject, details)) {
+
+            // When the variable is only declared but not initialized, `init` is `null`.
+            if (varInitAs && !this.allowedExpression(varInitAs, escapeObject, details)) {
```

Comment:
We can consider a variable with an `undefined` as an allowed identifier. 
Can you rewrite the patch such that if `varInitAs` is null, we can set `definedAsAllowed = true`?

Reply:
Ok, that sounds good to me. I didn't know what we wanted here so I thought I start a conversation with the smallest patch possible.

Reply:
@mozfreddyb @willdurand actually `defineAsAllowed` is already being set to `true` when `varInitAs` is going to be undefined (see at live 178):
https://github.com/mozilla/eslint-plugin-no-unsanitized/blob/02beff218f4acd60baf906076982f1bbca7f1260/lib/ruleHelper.js#L177-L178

But we are then looking for the write expression from the variable references, and given that there are no write references at all, we end up considering it disallowed because `allWritingRefsAllowed` is set to `false`.

I think that something like the following should make a non initialized `let` and `const` declaration to be allowed as @mozfreddyb did ask in his review comment:

```diff
diff --git a/lib/ruleHelper.js b/lib/ruleHelper.js
index 3a7f6e0..df14bfe 100644
--- a/lib/ruleHelper.js
+++ b/lib/ruleHelper.js
@@ -183,6 +183,12 @@ RuleHelper.prototype = {
             // the variable was declared as a safe value (e.g., literal)
             // now inspect writing references to that variable
             let allWritingRefsAllowed = false;
+
+            // With no write variable references, if it was defined as allowed then we should consider it safe.
+            if (variableInfo.references.filter(ref => ref.isWrite()).length === 0) {
+                allWritingRefsAllowed = true;
+            }
+
             for (const ref of variableInfo.references) {
                 // only look into writing references
                 if (ref.isWrite()) {
```